pixeltable/exprs/expr.py

```diff
@@ -699,78 +707,108 @@ def __rfloordiv__(self, other: object) -> 'exprs.ArithmeticExpr':
         return self._rmake_arithmetic_expr(ArithmeticOperator.FLOORDIV, other)
 
     def _make_string_expr(self, op: StringOperator, other: object) -> 'exprs.StringOp':
-        """
-        Make left-handed version of string expression.
-        """
-        from .literal import Literal
         from .string_op import StringOp
+        from .literal import Literal
 
-        if isinstance(other, Expr):
-            return StringOp(op, self, other)
-        if isinstance(other, typing.get_args(LiteralPythonTypes)):
-            return StringOp(op, self, Literal(other))
-        raise TypeError(f'Other must be Expr or literal: {type(other)}')
+        if not isinstance(other, Expr):
+            other_lit = Literal.from_object(other)
+            if other_lit is None:
+                op_symbol = str(op)
+                raise TypeError(
+                    f'ERROR during string/array operation ({op_symbol}): Operand must be an '
+                    'Expression or a compatible literal (string for +, int for *), '
+                    f'but received type {{type(other).__name__}}.'
```

Comment:
This is not correct. It will result in the exact string `{type(other).__name}` being shown to the user.